spec/Spec/Polynomials.cry

```diff
@@ -51,3 +52,18 @@ evaluate_polynomial_in_evaluation_form polynomial z =
             width' = toInteger width
             r = (pow_mod z' width') - 1
             evaluation = (results!0) * r * inverse_width
+
+/**
+ * The BLS basic data blob
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#custom-types
+ * Also @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#blob
+ */
+type BlsBlob = [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB][8]
+
+/**
+ * Convert a blob to list of BLS field scalars.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#blob_to_polynomial
+ */
+blob_to_polynomial : BlsBlob -> BlsPolynomial
+blob_to_polynomial blob = polynomial where
+    polynomial = [bytes_to_bls_field bytes | bytes <- (split`{FIELD_ELEMENTS_PER_BLOB} blob)]
```

Comment:
question: why is this split by `FIELD_ELTS_PER_BLOB` instead of `BYTES_PER_FIELD_ELT` as in the spec? Consider using `groupBy` instead (groupby and split are the same function with the type parameters in a different order. So if ```split`{x}``` gives you things of size `y`, that's equivalent to ```groupBy`{y}```)

Reply:
No good answer 😅 

I like your suggestion and will switch to match the spec.